packages/product/src/services/product-module-service.ts

```diff
@@ -369,6 +373,145 @@ export default class ProductModuleService<
     return JSON.parse(JSON.stringify(products))
   }
 
+  // TODO: cleanup, abstract common create and update steps
+  async update(
+    data: ProductTypes.UpdateProductDTO[],
+    sharedContext?: Context
+  ): Promise<ProductTypes.ProductDTO[]> {
+    const products = await this.baseRepository_.transaction(
+      async (manager) => {
+        sharedContext = sharedContext || {
+          transactionManager: manager,
+        }
+
+        const productVariantsMap = new Map<
+          string,
+          ProductTypes.CreateProductVariantDTO[]
+        >()
+        const productOptionsMap = new Map<
+          string,
+          ProductTypes.CreateProductOptionDTO[]
+        >()
+
+        const productsData = await Promise.all(
+          data.map(async (product) => {
+            const productData = { ...product }
+
+            if (!isDefined(productData.id)) {
+              throw new MedusaError(
+                MedusaError.Types.NOT_FOUND,
+                `"id" must be defined`
+              )
+            }
```

Comment:
suggestion(non-blocking):
```suggestion
            if (!isDefined(productData.id)) {
              throw new MedusaError(
                MedusaError.Types.NOT_FOUND,
                `Cannot update product without id`
              )
            }
```